queue/queue.go

```diff
@@ -0,0 +1,65 @@
+package queue
+
+import (
+	"context"
+	"time"
+
+	"github.com/lbryio/transcoder/db"
+	"github.com/lbryio/transcoder/formats"
+	"github.com/lbryio/transcoder/pkg/worker"
+	_ "github.com/mattn/go-sqlite3" // sqlite
+	"go.uber.org/zap"
+)
+
+var logger = zap.NewExample().Sugar().Named("queue")
+
+type Queue struct {
+	queries Queries
+}
+
+func NewQueue(db *db.DB) *Queue {
+	return &Queue{queries: Queries{db}}
+}
+
+func (q Queue) Add(url, sdHash, _type string) (*Task, error) {
```

Comment:
all the functions for this struct just pass the args on to `queries`. feels like a thin interface. is this struct necessary?

Reply:
This is the way sqlc generates code. I think the idea is to separate business logic from data retrieval, which I like. Not necessarily makes a whole lot of sense in each particular case.